addons/account/models/account_move_line.py

```diff
@@ -1161,9 +1161,14 @@ def _inverse_account_id(self):
     # CONSTRAINT METHODS
     # -------------------------------------------------------------------------
 
-    @api.constrains('account_id', 'journal_id', 'currency_id')
     def _check_constrains_account_id_journal_id(self):
+        """
+        Avoid using api.constrains here as in case of a write on
+        account move and account move line in the same operation, the check would be done
+        before all write are complete, causing a false positive
+        """
         for line in self.filtered(lambda x: x.display_type not in ('line_section', 'line_note')):
+            self.flush_recordset()
```

Comment:
Several stuff here:
1) shouldn't it be `line` instead of `self`?
2) Couldn't we just flush before the constrains check? (so we avoid flushing N times)